src/monitoring_tests/fee_monitoring/fee_monitoring.py

```diff
@@ -0,0 +1,242 @@
+"""
+Fee Monitoring
+"""
+
+import json
+from typing import Tuple
+import requests
+from src.monitoring_tests.template_test import TemplateTest
+from src.constants import SUCCESS_CODE, header
+from src.helper_functions import DecodedSettlement
+
+
+class FeeMonitoring:
+    """
+    Class for fee monitoring.
+    """
+
+    def get_order_execution(self, order, tx_hash) -> Tuple[int, int, int]:
```

Comment:
no need to be an instance method since self is not needed. Generally, I think it would be nice to collect logic that requests data from external APIs (e.g. AWS, our backend API, RPC, tenderly, Dune, etc) into separate classes and share their logic across tests.

Reply:
Those data requests are currently collected in the TemplateTest though more to avoid code duplication. Making this into a data collection class and/or splitting it into the different APIs would help.